moto/glue/models.py

```diff
@@ -248,6 +253,56 @@ def untag_resource(self, resource_arn, tag_keys):
         self.tagger.untag_resource_using_names(resource_arn, tag_keys)
 
 
+    #TODO: @Himani. Will Refactor validation logic as I find the common validation required for other APIs
+    def create_registry(self, registry_name, description, tags):
+        registry_name_pattern = re.compile(r"^[a-zA-Z0-9-_$#.]+$")
+        registry_description_pattern = re.compile(r"[\\u0020-\\uD7FF\\uE000-\\uFFFD\\uD800\\uDC00-\\uDBFF\\uDFFF\\r\\n\\t]*")
+
+        max_registry_name_length = 255
+        max_registries_allowed = 10
+        max_description_length = 2048
+        max_tags_allowed = 50
+
+        resource_length_message = "The resource name contains too many or too few characters."
+        illegal_resource_value_msg = "The parameter value contains one or more characters that are not valid."
+        max_registries_allowed_msg = "More registries cannot be created. The maximum limit has been reached."
+        registry_already_exists_msg = "Registry already exists. RegistryName: "
+        invalid_tag_params_msg = "Invalid Tag Parameters"
+        invalid_number_of_tags = "New Tags cannot be empty or more than 50"
+
+        if len(self.registries) >= max_registries_allowed:
+            error_message = max_registries_allowed_msg
+            raise ResourceNumberLimitExceededException("CreateRegistry", error_message)
+
+        elif registry_name == "" or len(registry_name.encode('utf-8')) > max_registry_name_length:
+            error_message = resource_length_message + " Parameter Name: registryName"
+            raise InvalidInputException("CreateRegistry", error_message)
```

Comment:
Code tends to be more readable with dedicated errors, so don't be afraid to create more exceptions.

This would be much more readable as:
`
if...
    raise NameTooLong
```
The exact error msg doesn't matter, that can be hidden away